app/src/main/java/com/cp/tasky/agenda/overview/data/repository/OverviewRepositoryImpl.kt

```diff
@@ -0,0 +1,23 @@
+package com.cp.tasky.agenda.overview.data.repository
+
+import com.cp.tasky.agenda.overview.data.remote.RemoteDataResource
+import com.cp.tasky.agenda.overview.domain.OverviewRepository
+import com.cp.tasky.agenda.overview.domain.model.LocalCacheOverview
+import com.cp.tasky.agenda.overview.domain.model.Overview
+import javax.inject.Inject
+
+class OverviewRepositoryImpl @Inject constructor(
+    remoteDataResource: RemoteDataResource
+): OverviewRepository {
+    override suspend fun getAgendaByDate(timeStamp: Long): Overview {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun syncAgendaRemotely(localAgenda: LocalCacheOverview) {
```

Comment:
It seems you plan to pass the local agenda to the repo to sync it. But it's the job of the repository to decide what needs to be synced how. So you'd instead retrieve the local agenda here with your local data source and sync that. Then you don't need this as a parameter.